### PR TITLE
Prompt session creation on execute if mismatch

### PR DESCRIPTION
**Pull request by broody**
Not implemented here is actually revoking previous session

image

### CODE REVIEW

File: packages/keychain/src/components/ConfirmTransaction.tsx [hunk 1]
```diff
@@ -23,14 +25,40 @@ export function ConfirmTransaction() {
...
+      })),
     [ctx.transactions],
   );
+  const updateSession = useMemo(() => {
+    if (policiesUpdated) return false;
+
+    const txnsApproved = transactions.every((transaction) =>
+      policies.some(
+        (policy) =>
+          addAddressPadding(policy.target) ===
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/ConfirmTransaction.tsx:45]: oh i didn't know about this util func. i should use that for formatAddress func.

File: packages/keychain/src/components/ConfirmTransaction.tsx [hunk 2]
```diff
@@ -23,14 +25,40 @@ export function ConfirmTransaction() {
...
+      ),
+    );
+
+    return !account.hasSession(ctx.transactions) && txnsApproved;
+  }, [transactions, policiesUpdated, policies, account, ctx.transactions]);
+
+  if (updateSession) {
+    return (
+      <CreateSession
+        isUpdate={true}
```
Comment (JunichiSugiura) [inline packages/keychain/src/components/ConfirmTransaction.tsx:57]: nit: you can omit `={true}`

**Review by JunichiSugiura — APPROVED** [review]
LGTM

**Comment by JunichiSugiura** [comment]
is there a case when game dev want to trigger just one tx with specific method(i.e. transfer) which doesn't want to include in session?

**Comment by broody** [comment]
> is there a case when game dev want to trigger just one tx with specific method(i.e. transfer) which doesn't want to include in session?

Yup, if the call is not in requested policies then we don't display Create Session and fallback to weabuthn execution.

https://github.com/cartridge-gg/controller/pull/719/files#diff-b27c08a79b37c2b11dfb61c56eee2b8e19659f140a75fa3c43d48f2d88c6f558R51

**Pull request merged**